tensorstore/kvstore/s3/s3_metadata_test.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstdlib>
+#include <cstring>
+#include <gtest/gtest.h>
+
+#include "tensorstore/kvstore/s3/s3_metadata.h"
+#include "tensorstore/util/status_testutil.h"
+
+
+using ::tensorstore::internal_kvstore_s3::FindTag;
+using ::tensorstore::internal_kvstore_s3::GetTag;
+using ::tensorstore::internal_kvstore_s3::TagAndPosition;
+using ::tensorstore::Result;
+
+namespace {
+
+/// Exemplar ListObjects v2 Response
+/// https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_ResponseSyntax
```

Comment:
Is is possible to get the response in json format? If not, maybe we should use a small xml parser such as ... (looking around) ... tinyxml2? expat? libxml?